database/sinkdb/state.go

```diff
@@ -34,25 +34,38 @@ func newState() *state {
 	}
 }
 
-// SetPeerAddr sets the address for the given peer.
-func (s *state) SetPeerAddr(id uint64, addr string) {
+// Peers returns the current set of peer nodes. The returned
+// map must not be modified.
+func (s *state) Peers() map[uint64]string {
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.peers[id] = addr
+	return s.peers
```

Comment:
I made this change so that Evict can lookup a node ID by address. Also, looking forward, I think we'll often need to grab the entire list of peers for things like broadcasting to the entire cluster. Since changes to the peer list should be infrequent, I think it's okay to make a new copy on every change.